tests/validation/tests/v3_api/test_istio.py

```diff
@@ -0,0 +1,273 @@
+        result = run_command(curl_cmd)
+        assert result == '<title>Simple Bookstore App</title>\n'
+    return result
+
+
+def create_and_test_booinfo_services(p_client, ns, timeout=DEFAULT_TIMEOUT):
+    book_info_file_path = ISTIO_PATH + '/bookinfo.yaml'
+    execute_kubectl_cmd('apply -f ' + book_info_file_path + ' -n '
+                        + ns.name, False)
+    result = execute_kubectl_cmd('get deployment -n' + ns.name, True)
+
+    for wl in result['items']:
+        wait_for_wl_by_id_to_active(p_client, wl['metadata']['uid'], 60)
+
+    rating_pod = execute_kubectl_cmd('get pod -l app=ratings -n' + ns.name)
+    assert len(rating_pod['items']) == 1
+
+    rating_pod_name = rating_pod['items'][0]['metadata']['name']
+    result = execute_kubectl_cmd(
+        'exec -it -n ' + ns.name + ' ' + rating_pod_name
+        + ' -c ratings -- curl productpage:9080/productpage'
+        + ' | grep -o "<title>.*</title>"', False)
+
+    start = time.time()
+    while result != '<title>Simple Bookstore App</title>\n':
+        if time.time() - start > timeout:
+            raise AssertionError(
+                "Timed out and failed to get bookinfo service ready")
+        time.sleep(.5)
+        result = execute_kubectl_cmd(
+            'exec -it -n ' + ns.name + ' ' + rating_pod_name
+            + ' -c ratings -- curl productpage:9080/productpage'
+            + ' | grep -o "<title>.*</title>"', False)
+        assert result == '<title>Simple Bookstore App</title>\n'
+    return result
+
+
+def wait_for_wl_by_id_to_active(client, workload_uuid,
+                                timeout=DEFAULT_TIMEOUT):
+    start = time.time()
+    workloads = client.list_workload(uuid=workload_uuid).data
+    wl = workloads[0]
+    while wl['state'] != "active":
+        if time.time() - start > timeout:
+            raise AssertionError(
+                "Timed out waiting for state to get to active")
+        time.sleep(.5)
+        workloads = client.list_workload(uuid=workload_uuid).data
+        assert len(workloads) == 1
+        wl = workloads[0]
+    return wl
+
+
+def create_and_verify_istio_app(p_client, ns, project, name):
+    answers = {
+        "enableCRDs": "true",
+        "gateways.istio-ingressgateway.type": "NodePort",
+        "gateways.istio-ingressgateway.ports[0].nodePort":
+            ISTIO_INGRESSGATEWAY_NODEPORT,
+        "gateways.istio-ingressgateway.ports[0].port": 80,
+        "gateways.istio-ingressgateway.ports[0].targetPort": 80,
+        "gateways.istio-ingressgateway.ports[0].name": "http2",
+    }
+    external_id = "catalog://?catalog=" + ISTIO_CATALOG + \
+                  "&template=" + ISTIO_TEMPLATE + \
+                  "&version=" + ISTIO_VERSION + \
+                  "&namespace=cattle-global-data"
+    print("creating istio catalog app")
+    app = p_client.create_app(
+        name=name,
+        externalId=external_id,
+        targetNamespace=ns.name,
+        projectId=project.id,
```

Comment:
Istio should be installed in the cluster's System project, as occurs when enabling manually from the UI. When istio is not deployed in the System project, graphs and metrics are not available within projects.

Reply:
updated, move the `istio apps` to be deployed in the `system` project and `istio-system` namespace.